src/pgduckdb_options.cpp

```diff
@@ -365,4 +373,101 @@ DECLARE_PG_FUNCTION(pgduckdb_recycle_ddb) {
 	PG_RETURN_BOOL(true);
 }
 
+void
+DuckdbRowSubscriptTransform(SubscriptingRef *sbsref, List *indirection, struct ParseState *pstate, bool isSlice,
+                            bool isAssignment) {
+	if (isAssignment) {
+		elog(ERROR, "Assignment to duckdb.row is not supported");
+	}
+
+	if (isSlice) {
+		elog(ERROR, "Creating a slice out of duckdb.row is not supported");
+	}
+
+	if (indirection == NIL) {
+		elog(ERROR, "Subscripting duckdb.row with an empty subscript is not supported");
+	}
+
+	if (list_length(indirection) != 1) {
+		/* TODO: Fix this, if the column contains an array we should be able to subscript that type */
+		elog(ERROR, "Subscripting duckdb.row is only supported with a single subscript");
+	}
+	pprint(indirection);
+
+	// Transform each subscript expression
+	foreach_node(A_Indices, subscript, indirection) {
+		Assert(!subscript->is_slice);
+		Assert(subscript->uidx);
+
+		Node *subscript_expr = transformExpr(pstate, subscript->uidx, pstate->p_expr_kind);
+		pprint(subscript_expr);
+		int expr_location = exprLocation(subscript->uidx);
+		Oid subscript_expr_type = exprType(subscript_expr);
+
+		Node *coerced_expr = coerce_to_target_type(pstate, subscript_expr, subscript_expr_type, TEXTOID, -1,
+		                                           COERCION_IMPLICIT, COERCE_IMPLICIT_CAST, expr_location);
+		if (!coerced_expr) {
+			ereport(ERROR, (errcode(ERRCODE_DATATYPE_MISMATCH), errmsg("duckdb.row subscript must have text type"),
+			                parser_errposition(pstate, expr_location)));
+		}
+
+		if (!IsA(subscript_expr, Const)) {
+			pprint(subscript_expr);
+			ereport(ERROR, (errcode(ERRCODE_DATATYPE_MISMATCH), errmsg("duckdb.row subscript must be a constant"),
+			                parser_errposition(pstate, expr_location)));
+		}
+		sbsref->refupperindexpr = lappend(sbsref->refupperindexpr, coerced_expr);
+	}
+
+	/* TODO: Trigger cache population, probably we should do this somewhere else */
+	pgduckdb::IsExtensionRegistered();
```

Comment:
Not sure where to put this though, this might very well be the first place we can actually hook into.